server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -460,20 +460,23 @@ protected Symbol visitExpression(Expression node, ExpressionAnalysisContext cont
 
         @Override
         protected Symbol visitCurrentTime(CurrentTime node, ExpressionAnalysisContext context) {
-            CurrentTime.Type type = node.getType();
-            if (!(type.equals(CurrentTime.Type.TIMESTAMP) || type.equals(CurrentTime.Type.TIME))) {
-                visitExpression(node, context);
-            }
             String funcName = CurrentTimestampFunction.NAME;
-            int defaultPrecision = CurrentTimestampFunction.DEFAULT_PRECISION;
-            if (type.equals(CurrentTime.Type.TIME)) {
-                funcName = CurrentTimeFunction.NAME;
-                defaultPrecision = CurrentTimeFunction.DEFAULT_PRECISION;
+            switch (node.getType()) {
+                case TIMESTAMP:
+                    break;
+
+                case TIME:
+                    funcName = CurrentTimeFunction.NAME;
+                    break;
+
+                default:
+                    visitExpression(node, context);
```

Comment:
btw. was this behavior to still fall-through to the remaining code even correct before? Looks like the code outside the switch would fail if the `default` case was hit.

Reply:
The missing case is ``DATE`` for scalar function ``current_date``, which at present is unsupported:

```
cr> select current_date;
SQLActionException[UnsupportedFeatureException: Unsupported expression current_date]
```

so the default will fail launch, throwing an exception:

```
 @Override
        protected Symbol visitExpression(Expression node, ExpressionAnalysisContext context) {
            throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
                "Unsupported expression %s", ExpressionFormatter.formatStandaloneExpression(node)));
        }
```